Octokit.Tests.Integration/Clients/UsersClientTests.cs

```diff
@@ -98,4 +98,17 @@ public async Task FailsWhenAuthenticatedWithBadCredentials()
             Assert.Equal(HttpStatusCode.Unauthorized, e.StatusCode);
         }
     }
+
+    public class TheGetAllMethod
+    {
+        [IntegrationTest]
+        public async Task ReturnsListOfdUsers()
+        {
+            var github = Helper.GetAuthenticatedClient();
+
+            var users = await github.User.GetAll("1");
+
+            Assert.Equal(30, users.Count);
```

Comment:
So while the other tests are looking good this one is the test I was waiting to dig into.

Because of how it's currently written, this API is only going to return the first page, with no options available for following the pagination. This doesn't really feel like a `GetAll` API, right? But we want to avoid this running forever, enumerating the millions of GitHub accounts, so how can we find this middle ground?

We have our existing pagination support so we could write the test like this:

```C#
[IntegrationTest]
public async Task ReturnsListOfUsers()
{
    var github = Helper.GetAuthenticatedClient();
    var options = new ApiOptions
    {
        PageSize = 10,
        PageCount = 2,
    };

    var users = await github.User.GetAll("1", options);

    Assert.Equal(20, users.Count);
}
```


However this API doesn't provide a `page=` parameter so if we used that it would just continue forever without returning any records to the user. We'd need to provide custom pagination logic for this case, as well as making more strict what the user can pass in with `ApiOptions`.